lib/acts_as_ar_query.rb

```diff
@@ -194,6 +194,49 @@ def append_hash_arg(symbol, *val)
     end
   end
 
+  def append_hash_array_arg(symbol, default, *val)
+    val = val.flatten.compact
+    val = val.first if val.size == 1 && val.first.kind_of?(Hash)
+    dup.tap do |r|
+      r.options[symbol] = merge_hash_or_array(r.options[symbol], val, default)
+    end
+  end
+
+  # @param a [Array, Hash]
+  # @param b [Array, Hash]
+  # @param default default value for conversion to a hash. e.g.: {} or "ASC"
+  def merge_hash_or_array(a, b, default = {})
+    if a.blank?
+      b
+    elsif b.blank?
+      a
+    elsif a.kind_of?(Array) && b.kind_of?(Array)
+      a + b
+    else
+      a = array_to_hash(a, default) if a.kind_of?(Array)
+      b = array_to_hash(b, default) if b.kind_of?(Array)
+      a.merge(b)
+    end
+  end
+
+  # This takes the array form and converts into the equivalent hash form
+  #
+  # examples:
+  #
+  # Vm.order(:name, :ip)
+  #   array_to_hash([:name, :ip], "ASC")
+  #   => {:name => "ASC", :ip => "ASC"}
+  #
+  # Vm.includes([:ext_management_systm, :host])
```

Comment:
Minor typo `ext_management_systm` =>` ext_management_system`

Also indent the examples line and the two actual example one indents worth... This way yard detects them as part of the description and not something else (alternately you can use the `@example` tag for each example